### PR TITLE
fixes killing of copy from, bulk insert, insert-by-subquery

### PR DESCRIPTION
**Pull request by seut**
this affects stopping of retries when kill is called

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/projectors/RetryListener.java [hunk 1]
```diff
@@ -56,6 +62,10 @@ public void onResponse(TResp response) {
 
     @Override
     public void onFailure(Exception e) {
+        if (executionFuture.isDone()) {
+            // if the execution future was completed (most likely failed or killed) we stop consuming
```
Comment (andreidan) [inline sql/src/main/java/io/crate/operation/projectors/RetryListener.java]: The comment should end with "we stop retrying" (not "consuming")

File: sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java [hunk 2]
```diff
@@ -326,9 +329,8 @@ private void processShardResponse(ShardResponse shardResponse) {
         return listener;
     }
 
-    private Map<ShardLocation, TReq> getFromPendingToRequestMap(
-        Map<String, List<PendingRequest<TItem>>> requestsByIndexForCurrentBulk) {
-        Map<ShardLocation, TReq> requests = new HashMap<>();
+    private void fromPendingToRequestMap(
```
Comment (andreidan) [inline sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java]: is this better? the method name and parameters are not expressive anymore ( method saying from something to something else but doesn't return anything)

File: sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java [hunk 3]
```diff
@@ -220,8 +220,11 @@ private ShardLocation getShardLocation(String indexName, String id, @Nullable St
                 nodeId = shardRouting.currentNodeId();
             }
 
-            if (nodeId == null && LOGGER.isDebugEnabled()) {
-                LOGGER.debug("Unable to get the node id for index {} and shard {}", indexName, id);
+            if (nodeId == null) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Unable to get the node id for index {} and shard {}", indexName, shardIterator.shardId());
+                }
+                return null;
```
Comment (andreidan) [inline sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java]: why are you returning null? Looking at createRowConsumer if the shardLocation is null we will add to pending requests and not to requests. Is that by design? If it is we need to document this somehow (it's a side effect now and not really sure it's correct).

File: sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java (hunk 2 above)
Comment (seut) [inline sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java]: yes right will rename

**Pull request merged**